addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3537,6 +3545,10 @@ const SnippetOptionWidget = Widget.extend({
      * @returns {Promise<string|undefined>|string|undefined}
      */
     _computeWidgetState: async function (methodName, params) {
+        const $target = params.target ? this.options.wysiwyg.getEditable().find(params.target) : this.$target;
+        if (!$target[0]) {
+            return;
+        }
```

Comment:
@Zinston You copy/paste that code in multiple generic methods, meaning their override would have to do the same, meaning that apparently-generic feature is only available to a bunch of generic methods, meaning you have to modify `this.$target` into `$target` in the whole implementation (breaking history)... That also complexifies the implementation of the method with a new "do I have the right `$target`?" at the beginning, things you should not have to care about when implementing an option method behavior (`selectStyle` = you want to force some style on the `$target`... that's it, not first asking yourself if you have the right `$target`).
Don't you think that option should have been coded on the caller side? That's what we do for the `apply-to` parameter... which is a near-duplicate of your new `target` parameter (one search in the whole DOM, one search in the original `$target`).

The minimum for me would to code it the same way as `apply-to` with a similar name. But what I think should be done is using the `apply-to` parameter as it is currently implemented and review the `this.$target` of your specific mass mailing option to be the whole editable area (maybe just a call to `setTarget` in the `init` would be enough?). Your new semi-generic parameter encourages a bad practice of having widgets displayed for a section A but changing a section B which could be anywhere in the page, far away from A => that's not the case for your specific option, the more the reason to code that specifically for your option by changing the base `$target` of your "Design" option as I suggest.

If you agree, could you make this a priority? I have small conflicts because of this, would be easier you change this first (otherwise I will be the one sending you conflicts by merging my resolved ones ^^).

cc @dmo-odoo

Reply:
@qsm-odoo I think you're right. We had discussed the fact that we probably would have to do something about this in a separate task (David left on vacation today and the freeze is next week so we had to leave that for later in order for the task to be ready on time). I'll start working on it asap so as to minimize issues for you. Sorry for the inconvenience.

Reply:
@qsm-odoo I'm a bit confused with this `apply-to` thingy. To compute the visibility of the option, in several places we do `this.$(params.applyTo).eq(0);` but in that context `params.applyTo` is a selector for an element in the editable area whereas `this.$el` is the option in the sidebar. So it never finds it. I tried to find a case in Website where it does and actually never got past the "if not then return". Is it broken or am I missing something?

Reply:
> in several places we do [...]

We use it at 3 places:
- The caller of option methods (e.g. `selectStyle`)
- The caller of `_computeWidgetState`
- The caller of `_computeWidgetVisibility`

> this.$(params.applyTo).eq(0); but in that context params.applyTo is a selector for an element in the editable area whereas this.$el is the option in the sidebar.

I think you misses the fact that `this.$(` does not use `this.$el` in options classes but `this.$target` as the source to search into (a bad decision that was taken a very long time ago) https://github.com/odoo/odoo/blob/ed530aef3c6d99712db055099b5f06a31e91d9cf/addons/web_editor/static/src/js/editor/snippets.options.js#L3297

> So it never finds it. I tried to find a case in Website where it does and actually never got past the "if not then return"

Seems to work on my side for the cases I tried, which case are you trying? :thinking:

Reply:
> I think you misses the fact that `this.$(` does not use `this.$el` in options classes but `this.$target` as the source to search into (a bad decision that was taken a very long time ago)

Aaaah thank you, that clarifies it!

> Seems to work on my side for the cases I tried, which case are you trying?

I had put a debugger in `updateUIVisibility` and tried clicking in various snippets, thinking I'd trigger it for the border option which uses `applyTo` but I guess I didn't try in the right places then ;)

Thanks for your help.